analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -18,116 +18,119 @@
-            binaryExpression.IsKind(SyntaxKind.LogicalOrExpression)
-                && HasAppropriateOperatorsForInequality(left, right)
-                && HasFloatingType(semanticModel, right.Left, right.Right);
-
-        private static bool IsIndirectEquality(SemanticModel semanticModel, BinaryExpressionSyntax binaryExpression, BinaryExpressionSyntax left, BinaryExpressionSyntax right) =>
-            binaryExpression.IsKind(SyntaxKind.LogicalAndExpression)
-                && HasAppropriateOperatorsForEquality(left, right)
-                && HasFloatingType(semanticModel, right.Left, right.Right);
-
-        private static bool HasFloatingType(SemanticModel semanticModel, ExpressionSyntax left, ExpressionSyntax right) =>
-            IsExpressionFloatingType(semanticModel, right) || IsExpressionFloatingType(semanticModel, left);
-
-        private static bool IsExpressionFloatingType(SemanticModel semanticModel, ExpressionSyntax expression) =>
-            IsFloatingPointNumberType(semanticModel.GetTypeInfo(expression).Type);
-
-        private static bool HasAppropriateOperatorsForEquality(BinaryExpressionSyntax left, BinaryExpressionSyntax right) =>
-            (left.OperatorToken.Kind() is SyntaxKind.GreaterThanEqualsToken && right.OperatorToken.Kind() is SyntaxKind.LessThanEqualsToken)
-            || (left.OperatorToken.Kind() is SyntaxKind.LessThanEqualsToken && right.OperatorToken.Kind() is SyntaxKind.GreaterThanEqualsToken);
-
-        private static bool HasAppropriateOperatorsForInequality(BinaryExpressionSyntax left, BinaryExpressionSyntax right) =>
-            (left.OperatorToken.Kind() is SyntaxKind.GreaterThanToken && right.OperatorToken.Kind() is SyntaxKind.LessThanToken)
-            || (left.OperatorToken.Kind() is SyntaxKind.LessThanToken && right.OperatorToken.Kind() is SyntaxKind.GreaterThanToken);
     }
+
+    private static string ProposedMessageForMemberAccess(SonarSyntaxNodeReportingContext context, ExpressionSyntax expression) =>
+        expression is MemberAccessExpressionSyntax memberAccess
+        && SpecialMembers.TryGetValue(memberAccess.GetName(), out var proposedMethod)
+        && context.SemanticModel.GetTypeInfo(memberAccess).ConvertedType is { } type
+        && IsFloatingPointType(type)
+            ? $"'{type.ToMinimalDisplayString(context.SemanticModel, memberAccess.SpanStart)}.{proposedMethod}()'"
+            : null;
+
+    private static string ProposedMessageForIdentifier(SonarSyntaxNodeReportingContext context, ExpressionSyntax expression) =>
+        expression is IdentifierNameSyntax identifier
+        && SpecialMembers.TryGetValue(identifier.GetName(), out var proposedMethod)
+        && context.SemanticModel.GetSymbolInfo(identifier).Symbol is { ContainingType: { } type }
+        && IsFloatingPointType(type)
+            ? $"'{proposedMethod}()'"
+            : null;
```

Comment:
I wouldn't do that. I took a look at it and it would make it less understandable

Reply:
Looks complex to make it simpler for me too: too many small differences to factorize it effectively:
- one uses `GetTypeInfo`, the other `GetSymbolInfo`
- one uses `ConvertedType`, the other `Symbol`, then gets its `ContainingType`, checking its non-nullability
- one makes use of the type in the generated string, calling `ToMinimalDisplayString`, the other doesn't

@martin-strecker-sonarsource if you have any suggestions (maybe using differently roslyin API), I'm keen to make the change.

Reply:
`GetTypeInfo` .. `ConvertedType` should also work for the identifier. So it might be possible to do it like so
`ProposedMessage(SemanticModel model, SyntaxNode(or ExpressionSyntaxNode) node, bool prefixType)` but I agree with Pavel that it might do more harm than good.

Reply:
@pavel-mikula-sonarsource , @martin-strecker-sonarsource : Since you both seem to agree, can this be resolved? 

Or an action on my part is expected?

Reply:
No action needed. The suggestion was optional in case you wanted to give it a try. That's why I left it open for better visibility.